src/snes/supermetroid/modules/RidleyOverkillTracker.js

```diff
@@ -0,0 +1,139 @@
+            memory.roomID.prevFrameValue !== undefined &&
+            memory.roomID.prevFrameValue !== Rooms.EMPTY &&
+            memory.roomID.value === Rooms.EMPTY
+        ) {
+            if (this.inRidleyFight) {
+                // were in ridley fight but reset due to ridley
+            }
+            this.reset();
+            return;
+        }
+
+        // if we are not yet in the fight, but we are in ridley room and he appears
+        if (
+            !this.inRidleyFight &&
+            memory.roomID.value === Rooms.LowerNorfair.RIDLEY_ROOM &&
+            this.checkChange(memory.enemyHP)
+        ) {
+            if (memory.enemyHP.value !== 0) {
+                this.inRidleyFight = true;
+                return;
+            }
+        }
+
+        // we are in ridley fight
+        else if (this.inRidleyFight) {
+
+            // we left ridley room in whatever direction
+            if (this.checkTransition(memory.roomID, Rooms.LowerNorfair.RIDLEY_ROOM)) {
+                if (this.additionalShotCounter > 0) {
+                    if (this.settings.displayDoorTimes.value) {
+                        sendEvent('ridleyOverkill', {shots:this.additionalShotCounter, damage:this.additionalDamageDealt});
+                    }
+                    else {
+                        sendEvent('msg', this.getMessage(), 3);
+                    }
+                }
+                this.reset();
+            }
+
+            // keep watching for the last hit
+            else if (memory.enemyHP.value <= 1000 && this.checkTransition(memory.enemyHP, undefined, 0)) {
+                this.startCounting = true;
+
+                // make sure to do next check only after 10 frames
+                this.lastCheck = Date.now();
+            }
+
+            // he is dead
+            else if (memory.enemyHP.value === 0 && this.startCounting) {
+
+                const time = Date.now();
+
+                // only check every 10 frames
+                if (time - this.lastCheck > (1000 / 60 * 10))  {
+                    if (memory.ridleyIframeTimer && memory.ridleyIframeTimer.value === 10) {
+                        this.additionalShotCounter++;
+
+                        if (memory.ridleyShotDamage && memory.ridleyShotDamage.value > 0) {
+                            // this is not reliable, somehow this is below 900 sometimes
+                            this.additionalDamageDealt += memory.ridleyShotDamage.value;
+                        }
+
+                        this.lastCheck = Date.now();
+                    }
+                }
+            }
+        }
+
+
+        // If samus gets hurt, we have to check if she's dead, so we can end the game
+        if (
+            this.inRidleyFight &&
+            this.checkChange(memory.samusHP) &&
+            memory.samusHP.value === 0 &&
+            memory.samusReserveHP.value === 0
+        ) {
+            this.reset();
+            this.reloadUnsafe = false;
+        }
+    }
```

Comment:
Again, using the value from the DeathEvent module we can shorten this code to use `globalState.isDeath`
```suggestion
        // If samus gets hurt, we have to check if she's dead, so we can end the game
        if (this.inRidleyFight && globalState.isDeath) {
            this.reset();
            this.reloadUnsafe = false;
        }
    }
```